### PR TITLE
docs: update "How is the `MessageShareCommitment` generated?"

### PR DESCRIPTION
**Pull request by rootulp**
Update payment docs to account for how message share commitment is generated post ADR 008.

Note: targets feature branch

### CODE REVIEW

File: x/payment/spec/docs.md [hunk 1]
```diff
@@ -2,9 +2,15 @@
...
+The payment module enables users to pay for arbitrary data to be published to the Celestia blockchain. Users create a single `MsgWirePayForData` transaction that is composed of:
 
-Further reading: [Message Block Layout](https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md)
+1. `Message`: the data they wish to publish
+2. `MessageNameSpaceId`: the namespace they wish to publish to
```
Comment (SweeXordious) [inline x/payment/spec/docs.md:8]:
```suggestion
2. `MessageNamespaceId`: the namespace they wish to publish to
```

File: x/payment/spec/docs.md [hunk 2]
```diff
@@ -2,9 +2,15 @@
...
+3. `MessageShareCommitment`: a signature and a commitment over their data when encoded into shares
+
+After the `MsgWirePayForData` transaction is submitted to the network, a block producer malleates their transaction into a `MsgPayForData` which doesn't include their data (a.k.a message). Both components get included in the data square in different namespaces: the `MsgPayForData` gets included in the transaction namespace and the associated data gets included in the namespace the user specified in the original `MsgWirePayForData`. Further reading: [Message Block Layout](https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md)
+
+After a block has been created, the user can verify that their data was included in a block via a message inclusion proof. A message inclusion proof the `MessageShareCommitment` in the original `MsgWirePayForData` and subtree roots of the block's data square to proove to the user that the shares that compose their original data do in fact exist in a particular block.
```
Comment (SweeXordious) [inline x/payment/spec/docs.md:13]:
```suggestion
After a block has been created, the user can verify that their data was included in a block via a message inclusion proof. A message inclusion proof uses the `MessageShareCommitment` in the original `MsgWirePayForData` and subtree roots of the block's data square to prove to the user that the shares that compose their original data do in fact exist in a particular block.
```

File: x/payment/spec/docs.md [hunk 3]
```diff
@@ -14,12 +20,8 @@ When a PayForData message is processed, it consumes gas based on the message siz
 
 ## Messages
 
-- [`MsgWirePayForData`](https://github.com/celestiaorg/celestia-app/blob/29e0a2751182499f7dc03598eabfc8d049ae62cb/x/payment/types/tx.pb.go#L32-L40) is a message that is created and signed by the user but it never ends up on-chain.
-- [`MsgPayForData`](https://github.com/celestiaorg/celestia-app/blob/29e0a2751182499f7dc03598eabfc8d049ae62cb/x/payment/types/tx.pb.go#L209-L219) is a "malleated" transaction that is created from metadata in the original `MsgWirePayForData`. `MsgPayForData` does end up on-chain.
-
-## PrepareProposal
-
-The malleation process occurs during the PrepareProposal step.
+- [`MsgWirePayForData`](https://github.com/celestiaorg/celestia-app/blob/b4c8ebdf35db200a9b99d295a13de01110802af4/x/payment/types/tx.pb.go#L32-L40) is created and signed by the user but it never ends up on-chain. Instead, it is "malleated" into it's component parts: data and a `MsgPayForData` message.
```
Comment (SweeXordious) [inline x/payment/spec/docs.md:23]:
```suggestion
- [`MsgWirePayForData`](https://github.com/celestiaorg/celestia-app/blob/b4c8ebdf35db200a9b99d295a13de01110802af4/x/payment/types/tx.pb.go#L32-L40) is created and signed by the user but it never ends up on-chain. Instead, it is "malleated" into its component parts: data and a `MsgPayForData` message.
```

**Review by SweeXordious — APPROVED** [review]
:+1: :+1: :+1:

**Comment by rootulp** [comment]
Thanks for proposing all the fixes @SweeXordious !

**Pull request merged**